Remotion/Web/ClientScript/Scripts/DatePicker.ts

```diff
@@ -0,0 +1,249 @@
+    frame.frameBorder = 'no';
+    frame.scrolling = 'no';
+    frame.style.width = '100%';
+    frame.style.height = '100%';
+    frame.marginWidth = "0";
+    frame.marginHeight = "0";
+
+    $(button).closest('div, td, th, body').append(datePicker);
+
+    if (DatePicker._datePicker_repositionTimer) 
+      clearTimeout(DatePicker._datePicker_repositionTimer);
+    var repositionHandler = function ()
+    {
+      if (DatePicker._datePicker_repositionTimer)
+        clearTimeout (DatePicker._datePicker_repositionTimer);
+
+      if (DatePicker._datePicker_current && DatePicker._datePicker_current.DatePicker == datePicker && datePicker.is (':visible'))
+      {
+        DatePicker.ApplyPosition (datePicker, $(button));
+        DatePicker._datePicker_repositionTimer = setTimeout (repositionHandler, DatePicker._datePicker_repositionInterval);
+      }
+    };
+
+    DatePicker.ApplyPosition (datePicker, $(button));
+    DatePicker._datePicker_repositionTimer = setTimeout(repositionHandler, DatePicker._datePicker_repositionInterval);
+
+    return datePicker;
+  }
+
+  public static ApplyPosition (datePicker: JQuery, button: JQuery): void
+  {
+    var datePickerLeft;
+    var datePickerTop;
+    var datePickerWidth;
+    var datePickerHeight;
+
+    var body = $ ('body');
+    var left = $(button).offset().left;
+    var top = $(button).offset().top;
+
+    //  Adjust position so the date picker is shown below 
+    //  and aligned with the right border of the button.
+    datePicker.css('left', Math.max (0, left - datePicker.width() + $(button).width()));
+    datePicker.css('top', Math.max (0, top + $(button).height()));
+    datePickerLeft = datePicker.offset().left;
+    datePickerTop = datePicker.offset().top;
+    datePickerWidth = datePicker.width();
+    datePickerHeight = datePicker.height();
+
+    //  Re-adjust the button, in case available screen space is insufficient
+    var visibleBodyTop = body.scrollTop();
+    var visibleBodyHeight = $(window).height();
+
+    var datePickerTopAdjusted = datePickerTop;
+    if (visibleBodyTop + visibleBodyHeight < datePickerTop + datePickerHeight)
+    {
+      var newTop = Math.max (0, $(button).offset().top - datePickerHeight);
+      if (newTop >= 0)
+        datePickerTopAdjusted = newTop;
+    }
+
+    var visibleBodyLeft = body.scrollLeft();
+    var visibleBodyWidth = $(window).width();
+
+    var datePickerLeftAdjusted = datePickerLeft;
+    if (datePickerLeft < visibleBodyLeft && datePickerWidth <= visibleBodyWidth)
+      datePickerLeftAdjusted = visibleBodyLeft;
+
+    datePicker.css('left', datePickerLeftAdjusted);
+    datePicker.css('top', datePickerTopAdjusted);
+  }
+}
+
+class DatePickerFrame {
+  public static Calendar_SelectionChanged(value: string): void
+  {
+    window.parent.DatePicker.UpdateValue(value);
+  }
+}
+
+function DatePicker_ShowDatePicker (button: string, container: HTMLElement, target: HTMLInputElement, src: string, width: number, height: number): void
```

Comment:
We only want those functions as global functions that are used from C# code. Is this the case for all functions?

And we only need those ex-global functions as public functions, that are used from the global functions (i.e. the C# code in Renderer and ControObjects. This may not be true for any and all global functions of re-motion JS files (see downstream-dependencies), but it is true for DatePicker.

Reply:
The global functions are only used by C#.

I don't quite get what you are trying to say with the second paragraph. The global functions are getting removed in a future task and the C# files will use the `DatePicker.ShowDatePicker` syntax instead. Are you saying that we cannot remove the functions because they are used downstream?

Reply:
No, I'm saying we only need those that are actually used in C# but the PR preserves all global functions, which I think contains unsed functions. -> please verify.

Reply:
Ok I removed most of the methods. Since we are changing the name of the function we are introducing breaking changes. Do you want to create breaking change tickets for each of the methods?

Reply:
Breaking Change: yes, please. Single ticket per type is sufficient (here, DatePicker / BocDateTimeValue). We should then list the changed/removed APIs in the ticket desription.

Reply:
What about making the static functions private when they're not used outside the script?

Reply:
`private` restricts access to the class itself so even the global functions could not access them.

Reply:
Yes. I was referring to the functions used only by the class itself.

Reply:
Which methods are not public API? All that are not used from C#?

Reply:
That is correct. See
> And we only need those ex-global functions as public functions, that are used from the global functions (i.e. the C# code in Renderer and ControObjects. This may not be true for any and all global functions of re-motion JS files (see downstream-dependencies), but it is true for DatePicker.

Reply:
Now I understand your previous comment 😄